sdcm/utils/benchmarks.py

```diff
@@ -0,0 +1,233 @@
+        if backend in ("aws", "aws-siren"):
+            return self._node.parent_cluster.params.get("instance_type_db")
+        elif backend in ("gce", "gce-siren"):
+            return self._node.parent_cluster.params.get("gce_instance_type_db")
+        else:
+            LOGGER.warning("Unrecognized backend type, defaulting to 'Unknown' for"
+                           "db instance type.")
+            return None
+
+    def _install_ubuntu_prerequisites(self):
+        package_list = ["make", "automake", "libtool", "pkg-config", "libaio-dev", "fio"]
+        try:
+            LOGGER.info("Installing Ubuntu prerequisites for the node benchmarks...")
+            for pkg in package_list:
+                self._node.install_package(pkg, wait_for_package_manager=False)
+            LOGGER.info("Ubuntu prerequisites for the node benchmarks installed.")
+        except Exception as exc:
+            LOGGER.warning("Failed to install Ubuntu prerequisites for the node benchmarking tools. "
+                           "Exception:\n%s", exc)
+
+    def _build_and_install_sysbench(self):
+        build_and_install_script = """\
+            cd ./sysbench
+            ./autogen.sh
+            ./configure --without-mysql
+            make -j
+            make install
+        """
+        try:
+            self._remoter.sudo(shell_script_cmd(build_and_install_script))
+        except Exception as exc:
+            LOGGER.warning("Failed to install sysbench from source. Exception:\n%s", exc)
+
+    def run_sysbench(self, thread_num: int = None, test_time: int = 120):
+        thread_num = thread_num or self._get_number_of_cores_on_node()
+        run_cmd = f"sysbench cpu --threads={thread_num} --time={test_time} run"
+        try:
+            sysbench_run_result = self._remoter.run(run_cmd)
+            results_dict = asdict(ScyllaNodeBenchmarkSysbenchResult(sysbench_run_result.stdout))
+            results_dict.pop("cmd_output")
+            self.benchmark_results.update(results_dict)
+        except Exception as exc:
+            LOGGER.warning("Failed to run sysbench on node %s with exception:\n%s", self.node_name, exc)
+
+    def _get_number_of_cores_on_node(self) -> int:
+        cat_cmd = "cat /proc/cpuinfo"
+        regex_pattern = re.compile(r"cpu cores\s+:\s+(?P<core_num>\d+)")
+        cat_result = self._remoter.run(cat_cmd).stdout
+        core_num = int(regex_pattern.search(cat_result).groupdict()["core_num"])
+        return core_num
+
+    def run_fio(self):
+        run_cmd = """\
+            cd cassandra-fio
+            ./fio_runner.sh lcs
+        """
+        self._remoter.sudo(shell_script_cmd(run_cmd), new_session=True, ignore_status=True)
+        self._get_fio_results()
+
+    def _get_fio_results(self):
+        fio_reports_path = "/home/ubuntu/cassandra-fio/reports/"
+        cat_cmd = f"cat {fio_reports_path}{'lcs.64k.fio.json'}"
+        try:
+            cat_out = self._remoter.run(cat_cmd)
```

Comment:
you can use ignore_status and don't use so wide try except. What do you think?

Reply:
Good idea in most places in this module (I fixed it in 2 places), but possibly not here, as it's not just the `cat` command that could fail, but also the following JSON manipulation. I think it's safer to wrap it in a `try / except` to avoid it crashing a test without good reason.